packages/edit-site/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -2,11 +2,22 @@
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
+import { useEntityProp } from '@wordpress/core-data';
 import { Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { wordpress } from '@wordpress/icons';
 
 function FullscreenModeClose( { icon } ) {
+	const [ siteIconURL ] = useEntityProp( 'root', 'site', 'site_icon_url' );
+	
+	const isRequestingSiteIcon = useSelect( ( select ) => {
+		return ! select( 'core/data' ).hasFinishedResolution(
+			'core',
+			'getEntityRecord',
+			[ 'root', 'site', undefined ]
```

Comment:
Doesn't this need to match the [`useEntityProp` args](https://github.com/WordPress/gutenberg/pull/22952/files#diff-5bb3dbd3c539d837e40778fd5c4e55baR11)?

Reply:
Great question. I'm not entirely sure because I couldn't find great documentation on `hasFinishedResolution`, but I'll investigate and update you here.

Reply:
one thought is that since `'root', 'site'` translates to a single query which fetches `/settings`, there probably isn't a need to be more granular, since every prop is resolved in the same request. (I think)